benchmark/clblast/blas3/gemm_batched.cpp

```diff
@@ -0,0 +1,221 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename gemm.cpp
+ *
+ **************************************************************************/
+
+#include "utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(std::string t1, std::string t2, int m, int k, int n,
+                     int batch_size) {
+  std::ostringstream str{};
+  str << "BM_Gemm<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << t1 << "/" << t2 << "/" << m << "/" << k << "/" << n << "/"
+      << batch_size;
+  return str.str();
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, ExecutorType* executorPtr, int t1, int t2,
+         index_t m, index_t k, index_t n, scalar_t alpha, scalar_t beta,
+         index_t batch_size, bool* success) {
+  // Standard test setup.
+  std::string t1s = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(t1));
+  std::string t2s = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(t2));
+  const char* t_a = t1s.c_str();
+  const char* t_b = t2s.c_str();
+
+  index_t lda = t_a[0] == 'n' ? m : k;
+  index_t ldb = t_b[0] == 'n' ? k : n;
+  index_t ldc = m;
+
+  // The counters are double. We convert m, n and k to double to avoid
+  // integer overflows and write them in the counters
+  double m_d = static_cast<double>(m);
+  double n_d = static_cast<double>(n);
+  double k_d = static_cast<double>(k);
+  double batch_size_d = static_cast<double>(batch_size);
+
+  state.counters["m"] = m_d;
+  state.counters["k"] = k_d;
+  state.counters["n"] = n_d;
+  state.counters["batch_size"] = batch_size_d;
+
+  {
+    double nflops_AtimesB = (2 * k_d - 1) * m_d * n_d;
```

Comment:
How does the FLOP calculation work with batches? Are K/M/N the overall size of the buffer, or the size of each individual matrix?

Reply:
I think they are sizes of each, not overall. And the Flops are multiplied by batch size on line 71, I think it's correct.

Reply:
Yes it's multiplied later, I thought it was clearer and avoids making mistakes when modifying this code to do the multiplication only once.